src/main/java/model/students/StudentsList.java

```diff
@@ -28,31 +39,72 @@
         return studentsMapFiltered;
     }
 
-    private StudentsList init() {
-        String[] firstNames = {"Olivia", "Amelia", "Isla", "Emily", "Ava", "Oliver", "Harry", "Jack", "George", "Noah"};
-        String[] lastNames = {"Smith", "Johnson", "Williams", "Jones", "Brown", "Davis", "Miller", "Wilson", "Moore", "Taylor"};
-        int[] ages = {18, 20, 21, 22, 18, 18, 19, 23, 23, 20};
-        int[] ids = {96, 97, 98, 99, 100, 101, 102, 103, 104, 105};
+    private void initFromFile() {
+        String fileData = readFromFile();
+
+        String[] lines = fileData.split(System.getProperty("line.separator"));
+        for (int i = 1; i < lines.length; i++) {
+            String[] data = lines[i].split(",");
+            String firstName = data[1];
+            String lastName = data[2];
+            String email = data[4];
+            String phone = data[5];
+
+            try {
+                checkContactData(email, phone);
+                Student st = new Student()
+                        .withId(Integer.parseInt(data[0]))
+                        .withFirstName(firstName)
+                        .withLastName(lastName)
+                        .withAge(Integer.parseInt(data[3]));
+                try {
+                    st.withEmail(email);
+                } catch (NotValidEmailException e) {
+                    System.out.println("Email of " + st.getFirstName() + " " + st.getLastName() + " is invalid");
+                    e.printStackTrace();
+                }
+                try {
+                    st.withPhone(phone);
+                } catch (NotValidPhoneException e) {
+                    System.out.println("Phone of " + st.getFirstName() + " " + st.getLastName() + " is invalid");
+                    e.printStackTrace();
+                }
+                studentsList.add(st);
+            } catch (NoContactDataException e) {
+                System.out.println("Contact data of " + firstName + " " + lastName + " is missed or invalid. The student data is not stored!!!");
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private String readFromFile() {
+        InputStream inputStream = StudentsList.class.getResourceAsStream(studentsDataPath);
+        StringBuilder text = new StringBuilder();
+        int charId;
 
-        for (int i = 0; i < 10; i++) {
-            studentsList.add(new Student(firstNames[i], lastNames[i], ids[i], ages[i]));
+        try {
+            while ((charId = inputStream.read()) != -1) {
+                text.append((char) charId);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
         }
-        return this;
+        return text.toString();
     }
 
     void printList() {
```

Comment:
it doesn't print but writes to file. Method should correspond to the thing it does.